### ISSUE TITLE
SELECT pglogical.replication_set_add_all_tables('default', ARRAY['public']); run forever

### ISSUE THREAD

**Issue by meotimdihia** [2ndQuadrant/pglogical#310]
Hi,  this query runs forever. And don't show any logs.
```SQL
SELECT pglogical.replication_set_add_all_tables('default', ARRAY['public']);
```
Do we have any way to debug this?

**Comment by eulerto** [comment]
Did you check the locks? This function acquires `SHARE ROW EXCLUSIVE` lock on each table in the replication set. Maybe there is another session that is preventing this lock to be granted. Check `pg_locks`.

**Comment by meotimdihia** [comment]
Sorry, I gave up and used native PostgreSQL pub-sub. Everything is done, currently, I can't debug this.

**Comment by osiro** [comment]
I had the same issue as you had.

Among all of the database tables, two of them were constantly in use by an application, causing locks and preventing `replication_set_add_all_tables` from running.

The solution I found was to add tables separately, one by one, and skip the locked ones:

```sql
SELECT pglogical.replication_set_add_table( set_name := 'default', relation := 'table1'::regclass );
SELECT pglogical.replication_set_add_table( set_name := 'default', relation := 'table2'::regclass );
SELECT pglogical.replication_set_add_table( set_name := 'default', relation := 'table3'::regclass );
...
```

After adding most of the other tables to the replication set and left the problematic ones aside, I monitored `pg_locks` and waited until the locked relations were released.

Finally, I pulled up all of the relations added to the replication set just to double-check if I haven't accidentally forgot anything.

```sql
SELECT * FROM pglogical.replication_set_table;
```
